### PR TITLE
Add attachment endpoints to OpenAPI

### PR DESCRIPTION
**Pull request by Concetta**
This PR adds Attachment endpoints to the OpenAPI as follows:

By user: 
- GET v2/users/:id/attachments * Lists the attachments in user
- POST v2/users/:id/attachments * Creates a new attachment

By transaction: 
- GET v2/transactions/:id/attachments * Lists the attachments assigned to the transaction
- POST v2/transactions/:id/attachments * Assign an attachment to the transaction
- DELETE v2/transactions/:transaction_id/attachments/:attachment_id * Unassign an attachment 

By attachment: 
- GET v2/attachments/:id * Find an attachment
- PUT v2/attachments/:id * Update the attachment title 
- DELETE v2/attachments/:id * Delete an attachment

### CODE REVIEW

File: openapi.json [hunk 1]
```diff
@@ -45,13 +45,35 @@
...
+              "categories.write": "Ability to edit and delete categories",
+              "attachments.read": "Access to view attachments",
+              "attachments.write": "Ability to create, update and delete attachments"
             }
           }
         }
       }
     },
     "responses": {
+      "201": {
```
Comment (maknz) [inline openapi.json]: @Concetta there's a few issues here:

* Both responses are using the Error schema, but they're not errors
* Since any data can be returned, we can't point to a particular schema for these. For the generic errors we have specified, the response is always an Error schema, so we can say that definitively.
* We don't need these; these globally defined responses are for common errors, not for all responses. Look at existing endpoints for 201 and 204 for how we respond with those, and remove these.

File: openapi.json [hunk 2]
```diff
@@ -2009,60 +2113,6 @@
               "type": "integer"
             },
             "example": 42
-          },
-          {
-            "name": "start_date",
```
Comment (maknz) [inline openapi.json:2014]: Why did you remove these query params?

File: openapi.json [hunk 3]
```diff
@@ -2736,6 +2786,387 @@
...
+        "summary": "Get attachment",
+        "description": "Gets a particular attachment by its ID.",
+        "tags": [
+          "Attachments"
+        ],
+        "parameters": [
+          {
+            "name": "id",
+            "in": "path",
+            "description": "Attachment's ID",
```
Comment (maknz) [inline openapi.json]: Follow the same wording we've been using already

> "description": "The unique identifier of the attachment.",

File: openapi.json [hunk 4]
```diff
@@ -2736,6 +2786,387 @@
...
+                  "file_name": {
+                    "type": "string",
+                    "description": "The file name of the attachment",
+                    "example": "taxi.png"
+                  },
+                  "file_data": {
+                    "type": "string",
+                    "format": "base64",
+                    "description": "the base64-encoded contents of the source file. The supported file types are png, jpg, pdf, xls, xlsx, doc, docx.",
+                    "example": "Base64.strict_encode64(where-my-file-is-kept)"
```
Comment (maknz) [inline openapi.json]: This example isn't valid Ruby nor is it pseudo-code really. I think we should just remove the example in this case.

File: openapi.json [hunk 5]
```diff
@@ -2736,6 +2786,387 @@
...
+          "404": {
+            "$ref": "#/components/responses/404"
+          },
+          "422": {
+            "$ref": "#/components/responses/422"
+          }
+        }
+      }
+    },
+    "/transactions/{id}/attachments/{attachment.id}": {
```
Comment (maknz) [inline openapi.json]: Is `attachment.id` correct? Should it not be `attachment_id`?

**Review by maknz — CHANGES_REQUESTED** [review]

File: openapi.json (hunk 2 above)
Comment (maknz) [reply openapi.json:2014]: From discussion in Slack: these were mistakenly documented for the create action so are just being cleaned up here.

Comment (Concetta) [reply openapi.json]: The query params were incorrectly added to the create action.

File: openapi.json (hunk 5 above)
Comment (Concetta) [reply openapi.json]: Updated to use transaction_id & attachment_id

**Review by maknz — APPROVED** [review]
Nice work. Uploaded to rdme :+1:
